spanpropagator.go

```diff
@@ -0,0 +1,78 @@
+package opentracing
+
+// SpanPropagator is responsible (a) for encoding Span instances in a manner
```

Comment:
this file is the most interesting/controversial piece of the PR... naming and semantics are both tricky here since nothing is symmetrical and we need to support models that create child spans across RPC boundaries and models that use a single span for both parties in an RPC.